src/NHibernate/Hql/Ast/ANTLR/QueryTranslatorImpl.cs

```diff
@@ -48,12 +49,14 @@ public partial class QueryTranslatorImpl : IFilterTranslator
 		/// <param name="parsedQuery">The hql query to translate</param>
 		/// <param name="enabledFilters">Currently enabled filters</param>
 		/// <param name="factory">The session factory constructing this translator instance.</param>
+		/// <param name="queryLoaderFactory">A query loader factory method.</param>
 		public QueryTranslatorImpl(
```

Comment:
As this constructor is public, an overload without the new parameter and adequate implementation for it to work properly must be kept, unless this change targets a new major version of NHibernate.

Reply:
```suggestion
				/// <summary>
		/// Creates a new AST-based query translator.
		/// </summary>
		/// <param name="queryIdentifier">The query-identifier (used in stats collection)</param>
		/// <param name="parsedQuery">The hql query to translate</param>
		/// <param name="enabledFilters">Currently enabled filters</param>
		/// <param name="factory">The session factory constructing this translator instance.</param>
		public QueryTranslatorImpl(
			string queryIdentifier,
			IASTNode parsedQuery,
			IDictionary<string, IFilter> enabledFilters,
			ISessionFactoryImplementor factory)
			: this(queryIdentifier, parsedQuery, enabledFilters, factory, new QueryLoaderFactory(), null)
		{
		}

		/// <summary>
		/// Creates a new AST-based query translator.
		/// </summary>
		/// <param name="queryIdentifier">The query-identifier (used in stats collection)</param>
		/// <param name="parsedQuery">The hql query to translate</param>
		/// <param name="enabledFilters">Currently enabled filters</param>
		/// <param name="factory">The session factory constructing this translator instance.</param>
		/// <param name="queryLoaderFactory">The query loader factory.</param>
		public QueryTranslatorImpl(
			string queryIdentifier,
			IASTNode parsedQuery,
			IDictionary<string, IFilter> enabledFilters,
			ISessionFactoryImplementor factory,
			IQueryLoaderFactory queryLoaderFactory)
			: this(queryIdentifier, parsedQuery, enabledFilters, factory, queryLoaderFactory, null)
		{
		}

		/// <summary>
		/// Creates a new AST-based query translator.
		/// </summary>
		/// <param name="queryIdentifier">The query-identifier (used in stats collection)</param>
		/// <param name="parsedQuery">The hql query to translate</param>
		/// <param name="enabledFilters">Currently enabled filters</param>
		/// <param name="factory">The session factory constructing this translator instance.</param>
		/// <param name="queryLoaderFactory">The query loader factory.</param>
		/// <param name="namedParameters">The named parameters information.</param>
		public QueryTranslatorImpl(
			string queryIdentifier,
			IASTNode parsedQuery,
			IDictionary<string, IFilter> enabledFilters,
			ISessionFactoryImplementor factory,
			IQueryLoaderFactory queryLoaderFactory,
			IDictionary<string, Tuple<IType, bool>> namedParameters)
		{
			_queryIdentifier = queryIdentifier;
			_stageOneAst = parsedQuery;
			_compiled = false;
			_shallowQuery = false;
			_enabledFilters = enabledFilters;
			_factory = factory;
			_queryLoaderFactory = queryLoaderFactory;
			_namedParameters = namedParameters;
		}
```

QueryLoaderFactory implements your suggested interface. This should ensure backwards compatibility.

Reply:
Ok.